app/controllers/pushes_controller.rb

```diff
@@ -43,44 +54,42 @@ def validate_ids
     # bad input
     return render json: {error: "There was a problem parsing your IDs. Please check your input and try again."} unless requested_ids && requested_ids.count > 0
 
-    query = ""
-    query += requested_ids.map { |id| %(id:#{id}) }.join(' OR ')
-    query = "(#{query})"
-
-    # use this builder so default one doesnt add fq to break our query!!
-    response, response_documents = search_results({}) do |builder|
-      # must pass in with .with here, search_results({q: ...}) is discarded
-      AMS::PushSearchBuilder.new(self).with({q: query, rows: 2147483647})
+    found_ids = []
+    requested_ids.each_slice(100).each do |segment|
+      query = build_query(segment)
+      found_ids += query_ids(query)
     end
 
-    found_ids_set = Set.new( response_documents.map(&:id) )
-    requested_ids_set = Set.new(requested_ids)
-
-    # get exclusive items, remove those exclusive to found_ids
-    missing_ids = (requested_ids_set ^ found_ids_set).subtract(found_ids_set)
+    missing_ids = verify_id_set(requested_ids, found_ids)
 
     all_valid = missing_ids.count == 0 ? true : false
     id_response = {all_valid: all_valid}
     id_response[:missing_ids] = missing_ids unless missing_ids.empty?
-    id_response[:id_query] = query
     render json: id_response
   end
 
-  def transfer_query
-    query_params = delete_extra_params(params)
-    query_params[:fl] = 'id'
+  def new
+    if params[:transfer] == 'true'
+      ze_params = params
+      query_params = delete_extra_params(ze_params)
```

Comment:
it is necessary, because the solr search is using `params` to make the search_params, but we still need the unmodified `params` to carry the `:id_field` parameter afterwards - I'll add the .dup!